### PR TITLE
stm32: rng enhancements

### PR DESCRIPTION
**Pull request by grevaillot**
Hi,

Here are a couple of patches i had on my tree regarding rng handling, adding a couple of helpers and handling potential errors in a nicer way, based on RM recomandations.

### CODE REVIEW

File: lib/stm32/common/rng_common_v1.c [hunk 1]
```diff
@@ -73,17 +87,29 @@ bool rng_get_random(uint32_t *rand_nr)
...
         uint32_t rv;
         bool done;
         do {
-                if (RNG_SR & RNG_SR_SECS) {
-                        rng_disable();
-                        rng_enable();
+
+                if (RNG_SR & RNG_SR_SEIS) {
+                        RNG_SR = RNG_SR & ~RNG_SR_SEIS;
+                        for (int i = 12; i != 0; i--)
```
Comment (karlp) [inline lib/stm32/common/rng_common_v1.c]: we still want {} here.

Comment (grevaillot) [reply lib/stm32/common/rng_common_v1.c]: my bad, fixed

File: lib/stm32/common/rng_common_v1.c [hunk 2]
```diff
@@ -94,10 +94,16 @@ uint32_t rng_get_random_blocking(void)
...
-                        rng_disable();
-                        rng_enable();
+
+                if (RNG_SR & RNG_SR_SEIS) {
+                        RNG_SR = RNG_SR & ~RNG_SR_SEIS;
+                        for (int i = 12; i != 0; i--) {
+                                rv = RNG_DR;
+                        }
+                        RNG_CR &= ~RNG_CR_RNGEN;
+                        RNG_CR |= RNG_CR_RNGEN;
```
Comment (karlp) [inline lib/stm32/common/rng_common_v1.c]: My only concern with this series is this one.  I'm really not sure that "both" is a good compromise.  I kinda feel like if you do the disable/enable, there was no point in the readout, and the readout of 12 is in the g0 manual as a speed up?  _or_ it's a different implementation? (more doubtful)  I just... feel weird doing both.

Comment (karlp) [reply lib/stm32/common/rng_common_v1.c]: eh, I'm not using the peripheral.  I'm going to go with your judgement on this.

**Comment by karlp** [comment]
merged all in e2ac1a63..a8a92b4c thanks.  still feels weird doing both approaches, but better than the prior code certainly, and you're using it, so that counts for more! thanks!

**Pull request closed without merging**